src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -3432,10 +3440,25 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 				const entries: QuickPickInput<TaskQuickPickEntryType>[] = [];
 				let selectedTask: Task | undefined;
 				let selectedEntry: TaskQuickPickEntryType | undefined;
-				this._showIgnoredFoldersMessage().then(() => {
+				this._showIgnoredFoldersMessage().then(async () => {
+					const { globGroupTasks } = await this._getGlobTasks(TaskGroup.Build._id);
+					let defaultTasks = globGroupTasks;
+					if (!defaultTasks?.length) {
+						defaultTasks = this._getDefaultTasks(tasks, false);
+					}
+					let defaultBuildTask;
+					if (defaultTasks.length === 1) {
+						const group: string | TaskGroup | undefined = defaultTasks[0].configurationProperties.group;
+						if (group) {
+							if (typeof group === 'string' && group === TaskGroup.Build._id) {
+								defaultBuildTask = defaultTasks[0];
+							} else if ((group as any)._id === TaskGroup.Build._id) {
```

Comment:
`group as TaskGroup` did not work here. not sure why

Reply:
Accessing privates like this will fail in production now due to variable mangling

Reply:
realized that check is unnecessary bc it's already filtered by group 

https://github.com/microsoft/vscode/blob/8bd293e87deb3d56598657a7b6207adf4e731d12/src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts#L3444